src/seismometer/controls/decorators.py

```diff
@@ -2,13 +2,79 @@
 """
 As for all of the utils subpackage, no other custom subpackages should be referenced.
 """
-import builtins
+import hashlib
+import logging
+import os
+import shutil
 from functools import wraps
+from inspect import signature
+from pathlib import Path
 from typing import Any, Callable
 
-from IPython.display import display
+from IPython.display import HTML
 from ipywidgets import Widget
 
+logger = logging.getLogger("seismometer")
+SEISMOMETER_CACHE_DIR = Path("./.seismometer_cache")
+SEISMOMETER_CACHE_ENABLED = os.getenv("SEISMOMETER_CACHE_ENABLED", "") != ""
+
+
+def disk_cached_html_segment(func: Callable[..., HTML]) -> HTML:
+    """
+    Caching decorator for HTML data
+
+    Parameters
+    ----------
+    func : Callable[..., HTML]
+        function that generates an HTML component
+
+    Returns
+    -------
+    HTML
+        displayable HTML component
+
+    Raises
+    ------
+    TypeError
+        If called with a function that does not annotate its return type as HTML, or that returns a non-HTML object.
+    """
+    sig = signature(func)
+    if sig.return_annotation != HTML:
+        raise TypeError("The function must return {HTML} object.")
+
+    if not SEISMOMETER_CACHE_ENABLED:
+        return func
+
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        arg_str = str((args, frozenset(kwargs.items())))
+        hash_object = hashlib.md5(arg_str.encode())
+        arg_hash = hash_object.hexdigest()
+
+        file_dir = Path("./.seismometer_cache") / "html" / func.__name__
+
+        filepath = file_dir / f"{arg_hash}.html"
+        if filepath and filepath.is_file():
+            logger.debug(f"From cache: {filepath}")
+            return HTML(filepath.read_text())
+        else:
+            result = func(*args, **kwargs)
+            if not isinstance(result, HTML):
+                raise TypeError("The function must return {HTML} object.")
+            file_dir.mkdir(parents=True, exist_ok=True)
+            filepath.write_text(result.data)
+        return result
+
+    return wrapped_func
+
+
+def _clear_cache_dir() -> None:
+    shutil.rmtree(SEISMOMETER_CACHE_DIR / "html", ignore_errors=True)
```

Comment:
I'm mostly sold on the extensibility of an html subdirectory, but can you finish convincing me that that is worth not fully cleaning up after ourselves?  
This function removes html but leaves the .seismometer_cache/ dir

Reply:
You can now clean up the whole cache dir by using DiskCachedFunction.clear_all().

I don't want to clear the cache since we can now re-use it between runs since the hashes for dataframes are now consistent.